pkg/operator/staticpod/controller/missingstaticpod/missing_static_pod_controller.go

```diff
@@ -0,0 +1,265 @@
+package missingstaticpodcontroller
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	corelisterv1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+type missingStaticPodController struct {
+	operatorClient                    v1helpers.StaticPodOperatorClient
+	podListerForTargetNamespace       corelisterv1.PodNamespaceLister
+	configMapListerForTargetNamespace corelisterv1.ConfigMapNamespaceLister
+	targetNamespace                   string
+	operandName                       string
+}
+
+// New checks the latest static pods pods and uses the installer to get the
+// latest revision.  If the installer pod was successful, and if it has been
+// longer than the terminationGracePeriodSeconds+10seconds since the installer
+// pod completed successfully, and if the static pod is not at the correct
+// revision, this controller will go degraded.  It will also emit an event for
+// detection in CI.
+func New(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	kubeInformersForTargetNamespace informers.SharedInformerFactory,
+	eventRecorder events.Recorder,
+	targetNamespace string,
+	staticPodName string,
+) factory.Controller {
+	c := &missingStaticPodController{
+		operatorClient:                    operatorClient,
+		podListerForTargetNamespace:       kubeInformersForTargetNamespace.Core().V1().Pods().Lister().Pods(targetNamespace),
+		configMapListerForTargetNamespace: kubeInformersForTargetNamespace.Core().V1().ConfigMaps().Lister().ConfigMaps(targetNamespace),
+		targetNamespace:                   targetNamespace,
+		operandName:                       staticPodName,
+	}
+	return factory.New().
+		ResyncEvery(time.Minute).
+		WithInformers(
```

Comment:
add `configMapListerForTargetNamespace` we need to make sure it has synced before we start using it. If you don't want to react to changes then add it to `WithBareInformers`